tests/integration/test_bundle.py

```diff
@@ -0,0 +1,139 @@
+        "-m",
+        ops_test.model_full_name,
+        bundle_path,
+    ]
+
+    retcode, stdout, stderr = await ops_test.run(*run_args)
+    assert retcode == 0, f"Deploy failed: {(stderr or stdout).strip()}"
+    logger.info(stdout)
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    """Deploy the autoscaling-model-server bundle."""
+    await cli_deploy_bundle(ops_test, bundle_path=BUNDLE_PATH)
+    # Configure knative-serving with Istio information
+    await ops_test.model.applications["knative-serving"].set_config(
+        {
+            "istio.gateway.name": ISTIO_GATEWAY_NAME,
+            "istio.gateway.namespace": ops_test.model.name,
+        }
+    )
+    await ops_test.model.wait_for_idle(status="active", timeout=1000, idle_period=90, raise_on_error=False)
+
+
+def test_inference_service_serverless_deployment(
+    ops_test: OpsTest, lightkube_client: lightkube.Client
+):
+    """Create an InferenceService and validate it has a status."""
+    # Use the model namespace for deploying the ISVC
+    serverless_namespace = ops_test.model.name
+
+    # Create InferenceService from example file
+    @tenacity.retry(
+        wait=tenacity.wait_exponential(multiplier=1, min=1, max=15),
+        stop=tenacity.stop_after_delay(30),
+        reraise=True,
+    )
+    def create_inf_svc():
+        lightkube_client.create(SKLEARN_INF_SVC_OBJECT, namespace=serverless_namespace)
+
+    # Assert InferenceService state is Available
+    @tenacity.retry(
+        wait=tenacity.wait_exponential(multiplier=1, min=1, max=15),
+        stop=tenacity.stop_after_attempt(30),
+        reraise=True,
+    )
+    def assert_inf_svc_state():
+        inf_svc = lightkube_client.get(
+            ISVC, SKLEARN_INF_SVC_NAME, namespace=serverless_namespace
+        )
+        conditions = inf_svc.get("status", {}).get("conditions")
+        for condition in conditions:
+            if condition.get("status") == "False":
+                status_overall = False
+                break
+            status_overall = True
+        assert status_overall is True
+
+    create_inf_svc()
+    assert_inf_svc_state()
+
+
+def test_perfom_inference(ops_test: OpsTest, lightkube_client: lightkube.Client):
+    """Perform a POST request with data for sklearn-iris ISVC."""
+    # This input data is hardcoded based on the example in https://kserve.github.io/website/latest/get_started/first_isvc/
+    sklearn_iris_input = dict(instances=[[6.8, 2.8, 4.8, 1.4], [6.0, 3.4, 4.5, 1.6]])
+    headers = {"Content-Type": "application/json"}
+    url = get_isvc_url(
+        isvc_name=SKLEARN_INF_SVC_NAME,
+        isvc_namespace=ops_test.model.name,
+        lightkube_client=lightkube_client,
+    )
+    inference_response = requests.post(
+        f"{url}/v1/models/{SKLEARN_INF_SVC_NAME}:predict",
+        headers=headers,
+        data=json.dumps(sklearn_iris_input),
+    ).text
+    assert inference_response == '{"predictions":[1,1]}'
```

Comment:
Are there utility functions (or even the whole test) that we could consider moving to `chisme`? I think a lot of this already exists in `kserve-operators` repository

Reply:
Yeah, I was thinking of something similar. The reality is that we could have an abstraction for performing inference, it can receive the headers, input data and URL path, perhaps. Not sure if this is a good candidate for chisme, though. It is very specific to kserve. Thoughts?

Reply:
Indeed, we do not have anything charm-specific in chisme. Although, I remember some discussions around adding `get_gateway_info()` there etc. It would be a useful helper in multiple places, but I don't know if it makes sense to have a separate package only for kserve helpers. You think a `kserve` [folder](https://github.com/canonical/charmed-kubeflow-chisme/tree/main/src/charmed_kubeflow_chisme) here does not align with `chisme` goals?

Reply:
> You think a kserve [folder](https://github.com/canonical/charmed-kubeflow-chisme/tree/main/src/charmed_kubeflow_chisme) here does not align with chisme goals?

I would disagree, chisme was originally designed to help with all the boilerplate code in multiple charms regardless their operation (i.e. their workloads). It's could also be quite a lot to install chisme (which is already a big pkg) just to import one test case. To me this is very specific to kserve and doesn't quite fit in chisme's generic solution.

I'm open to create a package for this, but I'd be interested in knowing what other commonalities are out there so we can fully consider the option.

Reply:
Ok, shall we discuss this with the team in order to evaluate the effort/benefits ratio?

Reply:
+1 on discussing with the team